maas/client/viscera/machines.py

```diff
@@ -174,6 +177,8 @@ class Machine(Object, metaclass=MachineType):
         :param hwe_kernel: The HWE kernel to deploy. Probably only relevant
             when deploying Ubuntu.
         :param comment: A comment for the event log.
+        :param wait: If specified, wait until the deploy is complete, polling
+            at the specified frequency.
```

Comment:
I think the wait is rather weird being the sleep interval. `wait` should be the maximum about of time to wait. So maybe:

`machine.deploy(wait=timedelta(mintues=10).total_seconds(), wait_interval=0.1)`

`wait_interval` should be optional with `wait` and should default to `5`. Checking every 100ms is very often and even every second is to fast. I think every 5 seconds is a good default.